sim/midas/src/main/cc/simif_f1.cc

```diff
@@ -80,19 +89,42 @@ void simif_f1_t::fpga_shutdown() {
   close(edma_read_fd);
 }
 
-void simif_f1_t::fpga_setup(int slot_id) {
-  /*
-   * pci_vendor_id and pci_device_id values below are Amazon's and available
-   * to use for a given FPGA slot.
-   * Users may replace these with their own if allocated to them by PCI SIG
-   */
-  uint16_t pci_vendor_id = 0x1D0F; /* Amazon PCI Vendor ID */
-  uint16_t pci_device_id =
-      0xF000; /* PCI Device ID preassigned by Amazon for F1 applications */
+/**
+ * Amazon PCI Vendor ID.
+ */
+constexpr uint16_t pci_vendor_id = 0x1D0F;
 
+/**
+ * Amazon PCI Device ID pre-assigned by for F1 applications.
+ */
+constexpr uint16_t pci_device_id = 0xF000;
+
+void simif_f1_t::fpga_setup(int slot_id, const std::string &agfi) {
   int rc = fpga_mgmt_init();
   check_rc(rc, "fpga_mgmt_init FAILED");
 
+  // If an AGFI was specified, re-load the image.
+  if (!agfi.empty()) {
+    fprintf(stderr, "Flashing AGFI: %s\n", agfi.c_str());
+
+    // Load the image.
```

Comment:
To ensure that the FPGA is loaded into a well-defined state, you should `fpga_mgmt_clear_local_image(slot_id)` before loading the image as suggested in https://github.com/aws/aws-fpga/blob/master/hdk/README.md#step5 and to align with the way the manager loads the AGFI.